tests/functional/test_pep517.py

```diff
@@ -1,5 +1,5 @@
 import pytest
-from pip._vendor import toml
+import toml
```

Comment:
This works!?

Reply:
Note that I added `toml` in test requirements. That because this file uses `dump` which `tomli` does not support.

Reply:
Oh I see you added `toml` to the test dependency, nevermind.

Reply:
Maybe add a comment saying that we're not using the vendored tomli library as that doesn't have write capability?

Reply:
:+1: added a comment

Reply:
Just an FYI: I just made [Tomli-W](https://github.com/hukkin/tomli-w) as a write-only counterpart to Tomli. It could be used here instead of `toml` if so desired.